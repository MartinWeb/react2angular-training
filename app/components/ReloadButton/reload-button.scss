button {
    padding: 10px;
    border-radius: 0;
    background-color: rgb(241, 90, 36);
    color: white;
    border: 1px solid rgb(241, 90, 36);

    &:hover {
        cursor: pointer;
    }
}