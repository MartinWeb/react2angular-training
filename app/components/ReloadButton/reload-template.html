<button ng-click="$ctrl.onReload()" ng-if="$ctrl.visibility">{{$ctrl.content}}</button>