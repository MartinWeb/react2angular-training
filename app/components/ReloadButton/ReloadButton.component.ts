import angular from "angular";
import template from './reload-template.html';
import './reload-button.scss';

export const ReloadButtonComponent: angular.IComponentOptions = {
    template,
    bindings: {
        onReload: '<',
        content: '@',
        visibility: '<'
    }
};