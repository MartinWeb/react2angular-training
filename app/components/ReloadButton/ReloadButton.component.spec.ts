import { ReloadButtonComponent } from "./ReloadButton.component";

describe('ReloadButtonComponent', () => {
    test('should have valid snapshot', () => {
        expect(ReloadButtonComponent.template).toMatchSnapshot();
    });

    test('should have valid bindings', () => {
        expect(ReloadButtonComponent.bindings).toMatchSnapshot();
    });
});