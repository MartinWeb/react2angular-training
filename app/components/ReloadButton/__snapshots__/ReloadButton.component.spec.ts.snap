// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReloadButtonComponent should have valid bindings 1`] = `
{
  "content": "@",
  "onReload": "<",
  "visibility": "<",
}
`;

exports[`ReloadButtonComponent should have valid snapshot 1`] = `"<button ng-click="$ctrl.onReload()" ng-if="$ctrl.visibility">{{$ctrl.content}}</button>"`;
